### PR TITLE
Update changelog for 3.0.0

### PR DESCRIPTION
**Pull request by pykong**
As `3.0.0` is going to be a big release I have substructured the section into user-facing changes ("features") and developer-facing ones ("other").

@Yajo Could you add the changes you have made leading up to the release of `3.0.0.`, so far?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,11 +3,30 @@
...
 
-- Dropped support for Python 3.5.
+This is a big release with many new features added and improved.
+The code base also received a lot of love and hardening.
+
+#### Features:
+
+- Minimal supported Python version is now 3.6.
 - Dropped support for deprecated `voodoo.json`.
+- Supporting self-templating.
```
Comment (Yajo) [inline CHANGELOG.md]: This hasn't landed yet.
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,11 +3,30 @@
...
+
+#### Other:
+
+- Moved to `poetry` for package management.
 - Type annotated entire code base.
+- Increased test coverage.
+- Ditched `ruamel.yaml` for `PyYaml`.
+- Ditched Travis CI for GitHub Actions.
+- Added `pre-commit` for enforced linting.
+- Added `black` and `isort` for code formatting.
```
Comment (Yajo) [inline CHANGELOG.md]:
```suggestion
- Added `prettier`, `black` and `isort` for code formatting.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,11 +3,30 @@
...
+
+#### Features:
+
+- Minimal supported Python version is now 3.6.
 - Dropped support for deprecated `voodoo.json`.
+- Supporting self-templating.
+- Introduced gitignore-style patterns for `exclude` und `skip-if-exists`.
+- Dropped support for `include` option.
+- Added support for extending content of config files via content of other files via `pyaml-include`.
+
```
Comment (Yajo) [inline CHANGELOG.md]:
```suggestion
- Customizable template extension.
- Ability to remember last answers.
- Template upgrades support, (based on the previous point) with migration tasks specification.
- Extended questions format, supporting help, format, choices and secrets.
- More beautiful prompts.
- New CLI experience.

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,11 +3,30 @@
...
+#### Other:
+
+- Moved to `poetry` for package management.
 - Type annotated entire code base.
+- Increased test coverage.
+- Ditched `ruamel.yaml` for `PyYaml`.
+- Ditched Travis CI for GitHub Actions.
+- Added `pre-commit` for enforced linting.
+- Added `black` and `isort` for code formatting.
+- Added `pytest` for running tests.
```
Comment (Yajo) [inline CHANGELOG.md]:
```suggestion
- Use `plumbum` as CLI and subprocess engine.

```

**Comment by Yajo** [comment]
IMHO you should consider using the squash and merge for things like this, to avoid the master branch full of meaningless commits.

**Comment by pykong** [comment]
@Yajo You are right. I was curious about GitHub's "merge review suggestion", never used that.
But sure normally I am a fan of squashing.

**Pull request merged**